_minutes/2022-02-03-wecg.md

```diff
@@ -0,0 +1,130 @@
+ 11. Carlos Jeurissen (Jeurissen Apps)
+ 12. Alexei (Privacy Badger)
+ 13. Krzysztof Modras (Ghostery)
+ 14. Mukul Purohit (Microsoft)
+ 15. SImeon Vincent (Google)
+ 16. Timothy Hatcher (Apple)
+
+
+## Meeting notes
+
+[Issue 151](https://github.com/w3c/webextensions/issues/151): webRequest API alignment and support in MV3
+
+ * [bradley] Raised this during the last meeting, e.g. perf issues. Hoping to align the behavior between all browsers, and get clarification on blocking webRequest. Mentioned specific use cases in the issue. One of the use cases could also be addressed by Mozilla's getFrameId proposal ([issue 12](https://github.com/w3c/webextensions/issues/12)), but it seems that Google has reservations against it.
+   * [timothy] Safari will support getFrameId.
+   * [simeon] The extension team is favorable towards this feature, but we need sign-off from other parts of Chrome.
+ * [timothy] webRequest is not supported in non-persistent background pages in Safari, because we looked at Chrome, which didn't support it either.
+   * [rob] This lack of support for blocking webRequest in event pages in Safari was also a question at [issue 148](https://github.com/w3c/webextensions/issues/148).
+ * [tomislav] Declarative API is preferred for cases where they can be used. We're implementing DNR for compatibility, and are keeping blocking webRequest for other cases.
+ * [giorgio] Would like to get more clarity on the concerns over the performance aspects of webRequest. Concerned about the lack of flexibility with DNR. Why does Chrome not take Mozilla's approach of supporting async listeners?
+ * [simeon] The declarative design originated from the desire to support functionality without broad host permissions. During the implementation we saw that the implementation is also useful for general use. Blocking webRequest is not viable in the long term. With many changes in MV3, we are paving the road for the future, even if their use is not clear now.
+ * [krysztof] Content blockers cannot work without host permissions, e.g. cosmetic filters require CSS in content pages. Removal of host permissions is a weak motivation for DNR.
+ * [alexei] Browser vendors seem to have the mistaken belief that they can enumerate all use cases and include that in the design of DNR. Privacy and security extensions need to continue to be able to respond quickly to evolving tracker techniques, which a declarative API inherently does not support. Additionally, when browser vendors talk about improving performance with DNR, these concerns go against real world experience where webRequest-based extensions already make browsing faster and more performant. Browser vendor performance improvement claims seem like solving for the wrong problem.
+ * [tomislav] I did not say that all use cases can be addressed by DNR, but that declarative APIs are preferable when they can be used.
+ * [oliver] Perhaps then both webRequest and DNR should remain available in all browsers.
```

Comment:
No hard feelings but I don't remember saying this - maybe it was someone else? I did say something along these lines:

> [oliver] One use case for the webRequest APIs at 1Password is filling basic auth. We'd still love to support this in MV3 but it could easily be solved by a new API that doesn't require access to web requests.

Reply:
According the version history, an anonymous contributor included this line between :33 and :37 after the start of the meeting. The message and tone/writing style seems to match with Alexei, so I'll attribute this statement to Alexei, and include your new sentence here.

@oliverdunk @ghostwords Could you confirm whether the revised version matches with what has been said during the meeting?

>  \* [alexei] Perhaps then both webRequest and DNR should remain available in all browsers.
> \* [oliver] One use case for the webRequest APIs at 1Password is filling basic auth. We'd still love to support this in MV3 but it could easily be solved by a new API that doesn't require access to web requests.

Reply:
Can't speak for Alexei's comment but fine by me!

Reply:
Apologies, it seems I misattributed the "Perhaps then both webRequest and DNR should remain available in all browsers" line. Was it @carlosjeurissen? I know somebody said it, but it wasn't recorded.

Reply:
@ghostwords Thanks for the mention! It was indeed me. What I said was something like the following:
"To make the discussion more productive, we can focus on keeping support for both webRequest and DNR side by side."

I can further elaborate on this for / during our next meeting.

Reply:
Thanks for clarifying, I have updated the meeting notes. The diff is at https://github.com/w3c/webextensions/compare/7508934114f637504cdcf063bad9dbe80f5d6350..e74b51bb652cf11a09487ad7eda9dc65dec6bfc6

Reply:
Thank you! Glad this is resolved.